### PR TITLE
Honor (don't ignore) /etc/kolibri/username when installing Kolibri — putting an end to 2 contradictory KOLIBRI_USER settings

### PR DESCRIPTION
**Pull request by holta**
This PR fixes the worst of several problems described on Sept 9, 2022 here:

- #115

Thanks @jredrejo @benjaoming @radinamatic for reviewing when you have time.

Recap: This fixes the serious problem of Kolibri's Debian/Ubuntu (apt / .deb) installer ignoring `/etc/kolibri/username`

Which results in Kolibri's 2 permanently contradictory settings (i.e. 2 contradictory values) for KOLIBRI_USER — i.e. in these 2 places:

- /var/cache/debconf/config.dat
- /etc/kolibri/username

Not pretty!  So a month or so later, typically the first time a school apt upgrades, Kolibri's new Debian package erroneously tries to force the school (and all https://internet-in-a-box.org schools, if they are online and run `apt update` etc) to try to change KOLIBRI_USER:

```
┌─────────────────────────────────────────────────────────┤ Kolibri configuration ├─────────────────────────────────────────────────────────┐
│ The default is to choose your preferred desktop user account, for instance to ensure access to importing data from external USB devices.  │
│                                                                                                                                           │
│ Entering a username that doesn't exist will create a new system user with home directory /var/<username>.                                 │
│                                                                                                                                           │
│ Which user account should own the Kolibri server?                                                                                         │
│                                                                                                                                           │
│ ubuntu   <<<< WRONG!!! (user "kolibri" is already specified at /etc/kolibri/username)   _________________________________________________ │
│                                                                                                                                           │
│                                                                  <Ok>                                                                     │
│                                                                                                                                           │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Why does this happen every single time a school first upgrades Kolibri?  One can only assume that Ansible (Linux automation, that [arranges](https://github.com/iiab/iiab/blob/master/roles/kolibri/tasks/install.yml) all Internet-in-a-Box installs) runs something like:

```
export DEBIAN_FRONTEND=noninteractive
apt install kolibri
```

So what happens is that Kolibri's Ubuntu/Debian package (1) Encodes the very misleading (erroneous, desktop-like) KOLIBRI_USER into /var/cache/debconf/config.dat (2) Effectively deferring its 3-to-5 debconf screens asking for things like KOLIBRI_USER until 1 or 2 months later, into the hands of the wrong person, after everything's installed in a school (3) Leading to serious confusion + conflicting KOLIBRI_USER settings down the road, that nobody wants 😖

CONCLUSION: This simple PR eliminates the conflicting settings, and at least brings us a lot closer to what the doc portrays at https://kolibri.readthedocs.io/en/latest/install/ubuntu-debian.html#id2 — pasted in here:

image

STRETCH GOAL FOR LATER: Eliminating all screens during `apt upgrade` would be far-and-away better for most schools, who are simply trying to upgrade Kolibri, and do **_not_** want to change any settings, or read long-winded technical screens 🥲 (as we all strive to help schools focus on learning, **_not_** unnecessary low-level administration!)

IN SHORT, Internet-in-a-Box can easily force far cleaner behavior during Kolibri upgrades (i.e. what schools prefer) with a hacky script like the following: _(but we'd rather work with Kolibri to solve the problem properly, thoughtfully and in due course, helping everyone)_

```
#!/bin/bash

#CONFIGFILE=/var/cache/debconf/config.dat
set -e

. /usr/share/debconf/confmodule
db_fset kolibri/user seen true
db_fset kolibri/init seen true
db_fset kolibri/init-instructions seen true
```

### CODE REVIEW

**Comment by holta** [comment]
BACKGROUND / CONTEXT: [/etc/kolibri/README](https://github.com/learningequality/kolibri-installer-debian/blob/master/debian/README.etc) puts repeated and very strong emphasis on file `/etc/kolibri/username` which seems very positive. ✅

While @benjaoming did most of that writing more than 4 years ago if course, I can only assume it's still largely on target.

To that end, let's all hope that this still-very-useful (though at this point error-prone and stale!) top-level `/etc/kolibri` doc can be refined now in 2023, to fix its [broken link](https://kolibri.readthedocs.io/en/latest/advanced.html)  and other small/diversionary errors 🏗️ 

e.g. Another small error in /etc/kolibri/README is the command it spells out to determine KOLIBRI_HOME ("this command will print it") though that doesn't work in most cases — with install like Internet-in-a-Box that take security seriously — as KOLIBRI_USER has no real / runnable shell in /etc/passwd :

- [iiab/iiab#3504](https://github.com/iiab/iiab/issues/3504#issuecomment-1482173242)

File: debian/kolibri.scripts-common [hunk 1]
```diff
@@ -346,15 +346,21 @@ kolibri_debconf_set_defaults()
         db_fset kolibri/pre-010-upgrade-system-user seen true
     fi
 
-    if [ "$USER_SEEN" == "false" ]
+    if [ -s /etc/kolibri/username ]    # Test that file exists with non-zero size
```
Comment (benjaoming) [inline debian/kolibri.scripts-common:349]: You can simplify this change quite a lot by adding the test to the list of tests in [line 341](https://github.com/learningequality/kolibri-installer-debian/pull/117/files#diff-c957f55a152335a0eab960eca7417081d595632ad57f1efca7a51c9fa4a83969R341) AFAICT

Comment (holta) [reply debian/kolibri.scripts-common:349]: I saw that earlier, yes.

But I chose to keep the change more localized & more legible.

i.e. to make clear this PR is basically a 2-line change wrapped in an if statement.

(@jredrejo is of course free to move things around as he sees fit!)

**Review by jredrejo — APPROVED** [review]
Despite I agree with @benjaoming comment, I'm approving this because the functionality is correct and I prefer not to delay it anymore

**Comment by holta** [comment]
I too agree with @benjaoming ✅

(PR was kept short to illustrate the core idea...)

**Comment by benjaoming** [comment]
Thanks for all the lovely mentions :heart: I want to mention back that I _do_ miss having time to work on this :heart:
